[Joey Chance] iP

joey-chance opened the pull request:
Duke
“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke frees your mind of having to remember things you need to do. It's,

text-based
easy to learn
FAST SUPER FAST to use
All you need to do is,

add your tasks.
let it manage your tasks for you 😉
And it is FREE!

Features:

 Managing tasks
 Managing deadlines (coming soon)
 Reminders (coming soon)

Commit:
Add Gradle support

joey-chance committed:
level-1

joey-chance committed:
level-2

joey-chance committed:
level-3

joey-chance committed:
level-4

joey-chance committed:
Added Automated Text UI Testing

joey-chance committed:
level-5

joey-chance committed:
Added remove function and handled more possible exceptions.

joey-chance committed:
edit expected.txt

joey-chance committed:
fix expected.txt

joey-chance committed:
level 7

joey-chance committed:
level 8

joey-chance committed:
fix conflicts between level-7 and level-8

joey-chance committed:
ignore duke.txt

joey-chance committed:
fix bugs

joey-chance committed:
use more OOP

joey-chance committed:
organize into packages

Ardentsoul commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,22 @@
+    }
+    @Override
+    public String toString() {
+        return "[D] " + super.toString() + " (by: " + date.format(DateTimeFormatter.ofPattern("MMM d yyyy")) +")";
```
Shouldn't there be a whitespace after the last "+" operator? 🤔
I have noticed this in your Event class too.

Ardentsoul commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    public static void main(String[] args) {
+        new Duke("./data","./data/duke.txt").run();
```
Shouldn't there a space character after the comma? 🤔

Ardentsoul commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,69 @@
+            Ui.showLine();
+            int id;
+            switch(input) {
+                case "list" :
```
Shouldn't the indentation for the cases in your switch block be aligned with the switch statement? 🤔

Ardentsoul commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,69 @@
+                    sc.reset();
+                    try {
+                        input = sc.nextLine().strip();
+                        if (input.isEmpty()) throw new DukeException("Do what, sir?");
```
Shouldn't there be Egyptian braces for your if statements? 🤔
I noticed this for your if statements in this class.

Ardentsoul commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,69 @@
+                        System.out.println(e.getMessage());
+                    }
+                    break;
+                case "mark" :
```
Shouldn't there be a comment stating that the missing break statement for your 'mark' and 'unmark' are intentional? 🤔

Ardentsoul reviewed (commented):
LGTM. Just a few nits to fix.

joey-chance committed:
add test for TaskList methods

joey-chance committed:
follow java coding standards

joey-chance committed:
add java docs

joey-chance committed:
add find method

joey-chance committed:
Merge javadoc and coding standard increments

joey-chance committed:
Merge level 9

justinekoh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,165 @@
+            while (scanner.hasNextLine()) {
+                String type = scanner.next();
+                String status = scanner.next();
+                String description = scanner.nextLine().strip();
+                switch (type) {
```
Perhaps you can consider adding a blank line here to separate the parsing code block and the matching code block :)

justinekoh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,165 @@
+    }
+
+    /**
+     * Creates a new Taks object which is then stored by being written to the todo list file.
```
```suggestion
     * Creates a new Task object which is then stored by being written to the todo list file.
```
Typo!

justinekoh reviewed (commented):
LGTM. Very clean and tidy, looks great!

ianfromdover commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+
+import java.util.Scanner;
+
+/**
+ * The Duke program implements a responsive application which keeps track of the user's to do list.
+ * The persona of Duke largely follows the J.A.R.V.I.S A.I from Marvel.
+ *
+ * @author joey-chance
+ * @version 1.0
+ * @since 2022-02-05
+ */
+public class Duke {
```
Perhaps you may want to rename the class to `JarvisAi` to suit the personality

ianfromdover commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+    /**
+     * Runs the Duke application.
+     */
+    public void run() {
+        Ui.greet();
+        String input;
+        Scanner sc = new Scanner(System.in);
+        input = sc.next();
+        Parser.parse(input, sc, tasks, storage);
+        Ui.bye();
+        sc.close();
+    }
```
Good use of the SLAP principle 👍

ianfromdover commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,101 @@
+                String[] event = input.split(" /at ");
+                try {
+                    if (event.length < 2) {
+                        throw new DukeException("Invalid event task, sir.");
```
Perhaps you might want to declare all the error messages as `public static final String`s in the DukeException class, and then `throw new DukeException(DukeException.INVALID_EVENT);` to make it more organised

ianfromdover commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,111 @@
+        if (id > tasks.size()) {
+            return;
+        }
+        switch (command) {
+        case "mark":
+            if (tasks.get(id - 1).isDone()) {
+                Ui.alreadyDone(tasks.get(id - 1));
+            } else {
+                tasks.get(id - 1).setDone();
+                Ui.done(tasks.get(id - 1).toString());
+            }
+            break;
+        case "unmark":
+            if (!tasks.get(id - 1).isDone()) {
+                Ui.alreadyNotDone(tasks.get(id - 1));
+            } else {
+                tasks.get(id - 1).setNotDone();
+                Ui.notDone(tasks.get(id - 1).toString());
+            }
+            break;
+        case "remove":
+            Ui.remove(tasks.remove(id - 1).toString());
+            break;
+        default:
+            Ui.unknownCommand(command);
+            break;
+        }
+    }
```
It might be a good idea to separate this into three distinct methods to be called from `Parser` instead :)

ianfromdover commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,101 @@
+                } finally {
+                    break;
+                }
+            case "mark":
+                // Fallthrough
+            case "unmark":
+                // Fallthrough
```
Perhaps you might want to call the methods `tasks.mark()` and `tasks.unmark()` to maintain the level of abstraction :)

ianfromdover reviewed (commented):
Well written and follows coding standards closely. Good use of descriptive naming for methods.

joey-chance committed:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d' into branch-A-Gradle

* commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d':
  Add Gradle support

joey-chance committed:
add gradle to project

joey-chance committed:
minor fixes as suggested by peers

joey-chance committed:
minor fixes to gradle

joey-chance committed:
no message

joey-chance committed:
add checkstyle

joey-chance committed:
add GUI

joey-chance committed:
minor fixes

joey-chance committed:
Remove unused methods

joey-chance committed:
Add assertions to avoid invalid input

There are several methods which require the input to be valid. This
includes requiring a Task to have a description, a Deadline to have
a description and a due date, an Event to have a description and a
venue and a valid task ID for update and get operations.

To do so, let’s include assert statements to ensure that these
conditions are met before we carry out the methods.

joey-chance committed:
Merge pull request #2 from joey-chance/branch-A-Assertions

Add assertions to avoid invalid input

joey-chance committed:
Improve code quality

There were still several breaches to the Java coding standard throughout
the codes. To fix them, let’s utilize the checkstyle to easily look for
these violations.

joey-chance committed:
Merge pull request #3 from joey-chance/branch-A-CodeQuality

Improve code quality

joey-chance committed:
Detect duplicate tasks

Users may forget that they have already added a task to their list and
thus may create duplicate tasks in the list. To avoid this, let’s add a
method to check if two Task objects are equal and another method
to check if the list already contains this task.

joey-chance committed:
Merge pull request #4 from joey-chance/branch-C-DetectDuplicates

Detect duplicate tasks

joey-chance committed:
minor fixes

joey-chance committed:
Ui screenshot

joey-chance committed:
no message

joey-chance committed:
Update README.md

joey-chance committed:
Update README.md

joey-chance committed:
Set theme jekyll-theme-slate

joey-chance committed:
no message

joey-chance committed:
Merge branches 'master' and 'master' of https://github.com/joey-chance/ip

* 'master' of https://github.com/joey-chance/ip:
  Set theme jekyll-theme-slate
  Update README.md
  Update README.md

* 'master' of https://github.com/joey-chance/ip:
  Set theme jekyll-theme-slate
  Update README.md
  Update README.md

joey-chance committed:
no message

joey-chance committed:
no message

joey-chance committed:
change from running Main to running Launcher

joey-chance committed:
Update README.md

joey-chance committed:
Include help command

joey-chance committed:
Merge pull request #5 from joey-chance/Help

Include help command

joey-chance committed:
Close application if user inputs "bye"

joey-chance committed:
Merge pull request #6 from joey-chance/Bye

Close application if user inputs "bye"
